WordPress/src/main/java/org/wordpress/android/ui/engagement/EngagedPeopleListFragment.kt

```diff
@@ -0,0 +1,213 @@
+import org.wordpress.android.ui.notifications.NotificationsDetailActivity
+import org.wordpress.android.ui.pages.SnackbarMessageHolder
+import org.wordpress.android.ui.reader.ReaderActivityLauncher
+import org.wordpress.android.ui.reader.tracker.ReaderTracker
+import org.wordpress.android.ui.utils.UiHelpers
+import org.wordpress.android.util.SnackbarItem
+import org.wordpress.android.util.SnackbarItem.Action
+import org.wordpress.android.util.SnackbarItem.Info
+import org.wordpress.android.util.SnackbarSequencer
+import org.wordpress.android.util.WPUrlUtils
+import org.wordpress.android.util.image.ImageManager
+import org.wordpress.android.viewmodel.ContextProvider
+import org.wordpress.android.viewmodel.observeEvent
+import javax.inject.Inject
+
+class EngagedPeopleListFragment : Fragment() {
+    @Inject lateinit var viewModelFactory: ViewModelProvider.Factory
+    @Inject lateinit var contextProvider: ContextProvider
+    @Inject lateinit var imageManager: ImageManager
+    @Inject lateinit var snackbarSequencer: SnackbarSequencer
+    @Inject lateinit var uiHelpers: UiHelpers
+    @Inject lateinit var readerTracker: ReaderTracker
+
+    private lateinit var viewModel: EngagedPeopleListViewModel
+    private lateinit var recycler: RecyclerView
+    private lateinit var loadingView: View
+    private lateinit var rootView: View
+    private lateinit var emptyView: ActionableEmptyView
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        (requireActivity().application as WordPress).component().inject(this)
+        viewModel = ViewModelProvider(this, viewModelFactory).get(EngagedPeopleListViewModel::class.java)
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        return inflater.inflate(R.layout.engaged_people_list_fragment, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        rootView = view
+        recycler = view.findViewById(R.id.recycler)
+        loadingView = view.findViewById(R.id.loading_view)
+        emptyView = view.findViewById(R.id.actionable_empty_view)
+
+        val listScenario = requireArguments().getParcelable<ListScenario>(KEY_LIST_SCENARIO)
+
+        val layoutManager = LinearLayoutManager(activity)
+
+        savedInstanceState?.getParcelable<Parcelable>(KEY_LIST_STATE)?.let {
+            layoutManager.onRestoreInstanceState(it)
+        }
+
+        recycler.layoutManager = layoutManager
+
+        initObservers()
+
+        viewModel.start(listScenario!!)
+    }
+
+    private fun initObservers() {
+        viewModel.uiState.observe(viewLifecycleOwner, { state ->
+            if (!isAdded) return@observe
+
+            loadingView.visibility = if (state.showLoading) View.VISIBLE else View.GONE
```

Comment:
We also have a `updateVisibility`, alongside `setTextOrHide` in `uiHelpers` :) But setting visibility in either way is fine 👍